client/ayon_usd/standalone/usd/pinning/pinning_file_helper.py

```diff
@@ -0,0 +1,286 @@
+        )
+        identifier_to_path_dict.update(recursive_result)
+
+    return identifier_to_path_dict
+
+
+# this function would work but in some UsdLib versions it will output <UDIM> tag and in some versions it will resolve ot 1001-1100. This function also dose not handle UDIM resolution at the moment
+# def get_asset_dependencies(
+#     layer_path: str, asset_identifier: str, print_debug: bool = False
+# ) -> Dict[str, str]:
+#     """will return a Dict[str, str] that holds the asset_identifier and the resolved path for a given usd file
+#     the file needs to be an existing real path
+#
+#     Args:
+#         layer_path: path to the resource
+#         asset_identifier: usd internal identifier for the resource
+#
+#     Returns: a flat dict containing all the asset_identifiers and resolved paths that are needed to construct the Usd-Stage
+#
+#     """
+#     if print_debug:
+#         print(f"UsdFile: {layer_path} ")
+#     if not layer_path:
+#         return {}
+#     ref_list = {}
+#
+#     if isinstance(layer_path, Ar.ResolvedPath):
+#         layer_path = layer_path.GetPathString()
+#
+#     ref_list[asset_identifier] = layer_path
+#     if not layer_path.endswith((".usd", ".usda", ".usdc")):
+#         return ref_list
+#     asset_paths = UsdUtils.ExtractExternalReferences(layer_path)
+#     flattened_reference_list = [item for sublist in asset_paths for item in sublist]
+#     if not flattened_reference_list:
+#         return ref_list
+#
+#     for ref in flattened_reference_list:
+#
+#         if os.path.isabs(ref) or is_uri(ref):
+#             resolved_path = resolver.Resolve(ref)
+#         else:
+#             parent_file_dir_name = os.path.dirname(layer_path)
+#             abs_path = os.path.normpath(os.path.join(parent_file_dir_name, ref))
+#             resolved_path = resolver.Resolve(abs_path)
+#
+#         if isinstance(resolved_path, Ar.ResolvedPath):
+#             resolved_path = resolved_path.GetPathString()
+#
+#         ref_list.update(get_asset_dependencies(resolved_path, ref))
+#     return ref_list
+
+
+def write_pinning_file(
+    output_path: str,
+    pinning_data: Dict[str, str],
+    create_missing_dirs: bool = True,
+    overwrite_ok: bool = True,
+) -> bool:
+    """writes out a pinning file to disk in the appropriate format
+
+    Args:
+        output_path: str path where the pinning file should be written to
+        pinning_data: str the pinning dict that holds all the pinning data
+        create_missing_dirs: bool
+        overwrite_ok: bool
+
+    Returns: bool Ture on successful write
+
+    """
+    # data validation
+    if not isinstance(pinning_data, dict):
+        print(f"pinning data is not a dict")
+        return False
+
+    if not output_path.endswith(".json"):
+        print(f"output_path is not a json")
+        return False
+
+    # file creatoin
+    if os.path.exists(output_path) and not overwrite_ok:
+        print(f"Output path does Exist {output_path}")
```

Comment:
Also, this should really be a raised error I'd say.
```suggestion
        print(f"Destination filepath already exists: {output_path}")
```